### PR TITLE
Move compression artifacts from rosbag2_cpp to rosbag2_compression

### PR DESCRIPTION
**Pull request by piraka9011**
This removes the dependency of both `zstd` and `rosbag2_compression` from `rosbag2_cpp`.

This is part of a multi-PR effort to:
* Create a reader that can decompress files/messages in `rosbag2_compression`.
* Create a writer that can compress files/messages in `rosbag2_compression`.
* Select the appropriate reader/writer to use in `rosbag2_transport` based on the CLI.
* Add CLI options for compression in `ros2bag`.

Signed-off-by: Anas Abou Allaban <allabana@amazon.com>

### CODE REVIEW

File: rosbag2_compression/src/rosbag2_compression/compression_options.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+{
+  switch (compression_mode) {
+    case CompressionMode::NONE:
+      return kCompressionModeNoneStr;
+    case CompressionMode::FILE:
+      return kCompressionModeFileStr;
+    case CompressionMode::MESSAGE:
+      return kCompressionModeMessageStr;
+    default:
+      ROSBAG2_COMPRESSION_LOG_ERROR_STREAM("CompressionMode not supported!");
```
Comment (zmichaels11) [inline rosbag2_compression/src/rosbag2_compression/compression_options.cpp]: Nit: We can actually log the numeric value of the unsupported `CompressionMode` here since `CompressionMode` is now a `uint32_t`

File: rosbag2_compression/test/rosbag2_compression/test_compression_options.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  EXPECT_EQ(compression_mode, rosbag2_compression::CompressionMode::FILE);
+}
+
+TEST(CompressionOptionsFromStringTest, MixedCaseMessageStringReturnsMessageMode)
+{
+  const std::string compression_mode_string{"MeSsAgE"};
+  const auto compression_mode = rosbag2_compression::compression_mode_from_string(
+    compression_mode_string);
+  EXPECT_EQ(compression_mode, rosbag2_compression::CompressionMode::MESSAGE);
+}
```
Comment (zmichaels11) [inline rosbag2_compression/test/rosbag2_compression/test_compression_options.cpp]: Either add a mixed-case test for `FILE` and `NONE` or expand this unit test to also test mixed-case forms of `FILE` and `NONE`.

File: rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp [hunk 3]
```diff
@@ -85,12 +60,22 @@ void SequentialReader::open(
...
-    open_storage();
-    setup_compression();
+
     file_paths_ = metadata_.relative_file_paths;
     current_file_iterator_ = file_paths_.begin();
+
+    storage_ = storage_factory_->open_read_only(
+      storage_options.uri, storage_options.storage_id);
+    if (!storage_) {
+      throw std::runtime_error("No storage could be initialized. Abort");
```
Comment (zmichaels11) [inline rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp]:
```suggestion
      throw std::runtime_error{"No storage could be initialized. Abort"};
```

File: rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp [hunk 4]
```diff
@@ -85,12 +60,22 @@ void SequentialReader::open(
...
+    }
+
   } else {
-    open_storage();
+    storage_ = storage_factory_->open_read_only(
+      storage_options.uri, storage_options.storage_id);
+    if (!storage_) {
+      throw std::runtime_error("No storage could be initialized. Abort");
```
Comment (zmichaels11) [inline rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp]:
```suggestion
      throw std::runtime_error{"No storage could be initialized. Abort"};
```

**Comment by piraka9011** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9049/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4799/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7397/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8978/)
